src/utils/color-helpers.js

```diff
@@ -0,0 +1,43 @@
+export const ALL_OTHERS_LABEL = "All Others"
+export const ALL_OTHERS_COLOR = "#888888"
+
+export function maybeUpdateDomainRange(
+  chart,
+  data,
+  dataAccessor,
+  domain,
+  range,
+  wrapKey = false
+) {
+  const dataArr = data.map(dataAccessor)
+  const domainRangeMap = new Map([...domain].map((key, i) => [key, range[i]]))
+  const dataMap = new Map(
+    [...dataArr].map((key, i) => [
+      key,
+      domainRangeMap.get(key) ??
+        chart.getColor(wrapKey ? { key0: key } : key, i)
+    ])
+  )
+  chart.customDomain([...dataMap.keys()])
+  chart.customRange([...dataMap.values()])
+}
+
+export function maybeUpdateAllOthers(chart, data, domain, range) {
+  if (
+    data.map(d => d.data.key0).includes(ALL_OTHERS_LABEL) &&
+    !domain.includes()
+  ) {
+    domain.push(ALL_OTHERS_LABEL)
+    range.push(ALL_OTHERS_COLOR)
+  } else if (
+    domain.includes(ALL_OTHERS_LABEL) &&
+    !data.map(d => d.data.key0).includes(ALL_OTHERS_LABEL)
```

Comment:
Data is looped through twice in this function, would be good to extract to variable at the top of the function.

Reply:
Good point, extracted both of those to the top of the function